### PR TITLE
[clang-format] Add PenaltyBreakScopeResolution option.

### LINKED ISSUE #78014: [clang-format] Ability to configure penalty for breaking scope resolution
Clang-format has many penalty options to be able to configure where line breaks are chosen to be placed.
There are also several hard-coded penalties in the code that are for specific situations that aren't worth making configurable.
I noticed that one such hard-coded penalty is for breaking at the scope resolution operator, though it appears that this would be straightforward and suitable as one of the configurable options.

### PR DESCRIPTION
**Pull request by rmarker**
Resolves #78014

### CODE REVIEW

**Review by rymiel — APPROVED** [review]

**Review by HazardyKnusperkeks — APPROVED** [review]

**Comment by rmarker** [comment]
Thanks, @rymiel, @HazardyKnusperkeks.
I don't know whether you want to wait to give more time for others to review.
In any case, I don't have write access, so it would be good if someone could merge this pull request when the time is right.
Thanks.

**Pull request merged**
